assembly/src/ast/parsers/mod.rs

```diff
@@ -217,3 +228,64 @@ fn parse_error_code(token: &Token, constants: &LocalConstMap) -> Result<u32, Par
         _ => Err(ParsingError::extra_param(token)),
     }
 }
+
+/// Parses a hexadecimal parameter value into a u64.
+///
+/// # Errors
+/// Returns an error if:
+/// - The length of a short hex string (big-endian) is not even.
+/// - The length of a short hex string (big-endian) is greater than 16.
+/// - The length of the chunk of a long hex string (little-endian) is not equal to 16.
+/// - If the string does not contain a valid hexadecimal value.
+/// - If the parsed value is greater than or equal to the field modulus.
+pub fn parse_hex_value(
```

Comment:
Does this function need to be public here? If it is used only in submodules, I'd keep it private.

Reply:
Indeed, my bad